### ISSUE TITLE
Add support for specifying a Google progressive web apps manifest

### ISSUE THREAD

**Issue by Richienb** [squidfunk/mkdocs-material#810]
## Description

We need to add an optional flag in the `mkdocs.yml` file called something like `manifest` which will allow users to customise or simply specify the location of their Google progressive web apps manifest file to be added.

This will solve the following issues:

- User will not be prompted to install the web app
- Is not configured for a custom splash screen
- Address bar does not match brand colours
- Name will be truncated when displayed on home screen

### Expected behavior

You should be able to specify the manifest file

### Actual behavior

There is no implementation at the moment other then manual theme file editing

### Package versions

* Python: 3.6.5
* MkDocs: 0.17.4
* Material: 2.9.0

### Project configuration

Unrelated, but here we go...

``` yaml
# Project Information
site_name: ROS Code Documentation
site_description: ROS Code is a revolutionary, modern and pleasing programming language to code in featuring a streamlined and speedy installation, a very straightforward and simplistic syntax and is provided as a Python 3 library.
site_author: Richie Bendall
site_url: https://www.ros-code.ga

# Repository
repo_name: Richienb/ROS-Code
repo_url: https://github.com/Richienb/ROS-Code
edit_uri: ""

# Copyright
copyright: 'Copyright &copy; 2017 - 2018 Richie Bendall'

# Configuration
theme:
  name: material
  palette:
    primary: 'indigo'
    accent: 'red'
  font:
    text: 'Open Sans'
    code: 'Roboto Mono'
  language: 'en'
  favicon: 'assets/images/favicon.ico'

# Page Tree
pages:
    - Home: index.md
    - Main:
      - Information Badges: badges.md
      - Commands: commands.md
    - Installation:
      - Install As Module: install-module.md
      - Install From Source: install-source.md
      - Install Version 1.0: install-old.md
    - Other:
      - Recommended Syntax: syntax.md
      - Makefile Commands: makefile-commands.md
      - API Usage: api-usage.md
    - Contribution:
      - Edit The Documentation: edit-docs.md
      - Edit The Source Code: edit-code.md
      - Packages Used: packages.md
      - Release Guidelines: release-guidelines.md
      - Unit Testing: unittesting.md
    - Version 1.0:
      - Home: versionone.md
      - Build Status: versionone-buildstatus.md
      - Enable: versionone-enable.md
      - Download And Install: versionone-install.md
      - Troubleshooting: versionone-troubleshooting.md
    - Fine Print:
      - Code Of Conduct: codeofconduct.md
      - License: license.md

# Customization
extra:
  social:
    - type: globe
      link: https://www.richie-bendall.ml
    - type: github-alt
      link: https://github.com/Richienb

extra_css:
  - css/details-support.css

extra_javascript:
  - js/details-support.js

# Extensions
markdown_extensions:
  - pymdownx.arithmatex
  - pymdownx.betterem:
      smart_enable: all
  - pymdownx.caret
  - pymdownx.critic
  - pymdownx.details
  - pymdownx.emoji
  - pymdownx.inlinehilite
  - pymdownx.magiclink
  - pymdownx.mark
  - pymdownx.smartsymbols
  - pymdownx.superfences
  - pymdownx.tasklist:
      custom_checkbox: true
  - pymdownx.tilde
  - markdown.extensions.tables
  - markdown.extensions.codehilite:
      guess_lang: false
  - markdown.extensions.def_list
  - markdown.extensions.footnotes
  - markdown.extensions.meta
  - markdown.extensions.toc:
      permalink: true
  - smarty
  - admonition
```

### System information

* OS: Windows 10.0.16299.64
* Browser: Chrome 67.0.3396.99

**Comment by squidfunk** [comment]
We could add an option to the `theme` or `extra` configuration in `mkdocs.yml` to specify the location. Could you provide a "best practice" Manifest which we could provide as a default if the user doesn't specify it or does it not make any sense? Brand colors were implemented in  f9298cf91f6ff88a90035dec1174b4c7093b5e8a for Android.

**Comment by Richienb** [comment]
Sure, I'll add a best practice in the following comment.

**Comment by Richienb** [comment]
Here is my first draft on the default manifest
```yml
{
  "short_name": "Docs Name", # The name, but with a character limit of 12 to fit on most home screens
  "name": "Documentation Name", # The full name

  "icons": [{
    {
      "src": "/assets/images/favicon.ico", # Set default favicon if already specified in mkdocs.yml file
      "sizes": "32x32", # Size of the favicon
      "type": "image/png"
    }
  ],
  "background_color": "#FAFAFA", # Set background colour, we should go with either #FFFFFF or #FAFAFA
  "display": "standalone", # Launches window in full screen
  "theme_color": "#3F51B5", # Another way to set the theme colour
  "start_url": "/?utm_source=homescreen" # Set start url. The reason for ?utm_source=homescreen is to be compatible with Google Analytics and will still work if disabled
}
```

**Comment by Richienb** [comment]
Other options which can be specified are (examples are shown)
```yml
"orientation": "landscape" # Force screen orientation

"scope": "/maps/" # The part of the website that uses the "app"

"prefer_related_applications": true, # Suggest an app already on Play Store
"related_applications": [
  {
    "platform": "play",
    "id": "com.google.samples.apps.iosched"
  }
]
```

**Comment by Richienb** [comment]
You can also check out the manifest I'm using for my website [here](https://cdn.rawgit.com/Richienb/richienb.github.io/development/manifests/manifest.json)

**Comment by squidfunk** [comment]
Thanks! However, we cannot provide a default configuration since nearly everything depends on configuration. For this reason we can only provide a path to the `manifest.json` but that should be enough. Will implement this when I find the time!

**Comment by Richienb** [comment]
Ok. Thanks!

**Comment by Richienb** [comment]
Just wondering, has this feature been implemented yet?

**Comment by squidfunk** [comment]
Nope, haven't found the time yet, I'm sorry.

**Comment by Richienb** [comment]
Ok No problem. :stuck_out_tongue:

**Comment by squidfunk** [comment]
I finally founds some time to tackle this. Adding a default is actually not possible, because Jinja cannot parse JSON files, so we're not able to use variables inside JSON files. However, I added a simple option `config.extra.manifest` in the latest commit (see master) which should point to your Manifest contained in yout `docs` folder. Does this solve your problem? If so, we need to add some documentation on it and can make a release within the next days.

**Comment by Richienb** [comment]
Yes! This is what I was looking for. I've opened pull request https://github.com/squidfunk/mkdocs-material/pull/916 which adds the doumentation for this feature.

**Comment by squidfunk** [comment]
Released as part of 3.1.0. Thanks for your patience and support!
